creator-node/src/services/stateMachineManager/stateReconciliation/SecondarySyncHealthTracker.ts

```diff
@@ -1,332 +1,301 @@
-  date?: string | null
-}
+      // Set the expiry by REDIS_KEY_EXPIRY_SECONDS
+      await redisClient.expire(redisKey, REDIS_KEY_EXPIRY_SECONDS)
 
-/**
- * Given a mapping of wallet to secondaries arrays, returns mapping from wallet to
- * sync metrics for that wallet, where sync metrics are a mapping of secondary endpoint
- * to successCount, failureCount, and successRate
- *
- * @param {Object { <wallet (string)>: <secondary endpoints (string array)}} walletsToSecondariesMapping
- * @returns {Object} { '0x...': { 'https://secondary1...': { 'successCount' : _, 'failureCount': _, 'successRate': _ }, ... } ... }
- */
-export async function computeUsersSecondarySyncSuccessRatesForToday(
-  walletsToSecondariesMapping: WalletsToSecondariesMapping
-): Promise<UserSecondarySyncMetricsMap> {
-  // Initialize sync success and failure counts for every secondary to 0
-  const secondarySyncMetricsMap: UserSecondarySyncMetricsMap = {}
-  const wallets = Object.keys(walletsToSecondariesMapping)
-  for (const wallet of wallets) {
-    const secondarySyncMetrics = secondarySyncMetricsMap[wallet] || {}
-    for (const secondary of walletsToSecondariesMapping[wallet]) {
-      secondarySyncMetrics[secondary] = {
-        successCount: 0,
-        failureCount: 0,
-        successRate: 1
-      }
-      secondarySyncMetricsMap[wallet] = secondarySyncMetrics
+      genericLogger.debug(
+        { SecondarySyncHealthTracker: 'recordFailure' },
+        `Incremented ${redisKey}:${prometheusError} count`
+      )
+    } catch (e: any) {
+      genericLogger.error(
+        { SecondarySyncHealthTracker: 'recordFailure' },
+        `Failed to increment ${redisKey}:${prometheusError} count`
+      )
     }
   }
 
-  // Retrieve map of all SyncRequestOutcome keys and daily counts for wallets from all secondaries
-  if (PROCESS_SYNC_RESULTS) {
-    const userSecondarySyncHealthOutcomes =
-      await _batchGetSyncRequestOutcomeMetricsForToday(wallets)
-
-    // Aggregate all daily SyncRequest outcome counts by secondary
-    _aggregateSyncSuccessAndFailureCountBySecondaryPerWallet(
-      userSecondarySyncHealthOutcomes,
-      secondarySyncMetricsMap
-    )
-
-    // For each secondary, compute and store successRate
-    _aggregateSyncSuccessRateBySecondaryPerWallet(
-      wallets,
-      secondarySyncMetricsMap
-    )
-  }
-
-  return secondarySyncMetricsMap
-}
+  /**
+   * If there is no presence of the wallet, wallet-secondary pair, that means no
+   * error was encountered. Else, return what is in the map
+   * @param wallet
+   * @param secondary
+   * @returns flag on whether wallet on secondary exceeded the max attempts allowed
+   */
+  doesWalletOnSecondaryExceedMaxErrorsAllowed(
+    wallet: string,
+    secondary: string
+  ): boolean {
+    if (
+      !this.walletToSecondaryToExceedsMaxErrorsAllowed ||
+      !this.walletToSecondaryToExceedsMaxErrorsAllowed[wallet] ||
+      !this.walletToSecondaryToExceedsMaxErrorsAllowed[wallet][secondary]
```

Comment:
nit - can we change this to a cleaner one-liner: `!this.walletToSecondaryToExceedsMaxErrorsAllowed?.[wallet]?.[secondary]`